docs/genai/api/java.md

```diff
@@ -0,0 +1,595 @@
+```
+
+### setInput Method
+
+Sets the prompt/s token ids for model execution. The `tokenIds` are the encoded parameters.
+
+```java
+public void setInput(int[] tokenIds, int sequenceLength, int batchSize)
+ throws GenAIException
+```
+
+#### Parameters
+
+- `tokenIds`: the token ids of the encoded prompt/s
+- `sequenceLength`: the length of each sequence.
+- `batchSize`: size of the batch. 
+
+#### Throws
+
+`GenAIException`- if the call to the GenAI native API fails. 
+
+NOTE: all sequences in the batch must be the same length.
+
+#### Example
+
+```java
+generatorParams.setInput(tokenIds, sequenceLength, batchSize);
+```
+
+## Generator class
+
+The Generator class generates output using a model and generator parameters.
+The expected usage is to loop until isDone returns false. Within the loop, call computeLogits followed by generateNextToken.
+
+The newly generated token can be retrieved with getLastTokenInSequence and decoded with TokenizerStream.Decode.
+
+After the generation process is done, GetSequence can be used to retrieve the complete generated sequence if needed.
+
+### Create a Generator
+
+Constructs a Generator object with the given model and generator parameters.
+
+```java
+Generator(Model model, GeneratorParams generatorParams)
+```
+
+#### Parameters
+
+- `model`: the model.
+- `params`: the generator parameters.
+
+#### Throws
+
+`GenAIException`- if the call to the GenAI native API fails.
+
+### isDone Method
+
+Checks if the generation process is done.
+
+```java
+public boolean isDone()
+```
+
+#### Returns
+
+Returns true if the generation process is done, false otherwise.
+
+### computeLogits Method
+
+Computes the logits for the next token in the sequence.
+
+```java
+public void computeLogits() throws GenAIException
+```
+
+#### Throws
+
+`GenAIException`- if the call to the GenAI native API fails.
+
+### getSequence Method
+
+Retrieves a sequence of token ids for the specified sequence index.
+
+```java
+public int[] getSequence(long sequenceIndex) throws GenAIException
+```
+
+#### Parameters
+- `sequenceIndex`: the index of the sequence.
+
+#### Throws
+
+`GenAIException`- if the call to the GenAI native API fails.
+
+#### Returns
+
+An array of integers with the sequence of token ids.
+
+#### Exmaple
+```java
+int[] outputIds = output.getSequence(i);
+```
+
+### generateNextToken Method
+
+Generates the next token in the sequence.
+
+```java
+public void generateNextToken() throws GenAIException
+```
+
+#### Throws
+
+`GenAIException`- if the call to the GenAI native API fails.
+
+### getLastTokenInSequence Method
+
+Retrieves the last token in the sequence for the specified sequence index.
+
+```java
+public int getLastTokenInSequence(long sequenceIndex) throws GenAIException
+```
+
+#### Parameters
+
+- `sequenceIndex`: the index of the sequence.
+
+#### Throws
+
+`GenAIException`- if the call to the GenAI native API fails.
+
+#### Returns
+
+The last token in the sequence.
+
+## Sequence Class
```

Comment:
Need a constructor for the Sequences class

Reply:
Also Sequence -> Sequences

Reply:
fixed sequence -> sequences. Added getSequence method. Didn't add constructor because there is no public way to generate a sequences handle (the parameter).